include/mbedtls/ccm.h

```diff
@@ -107,12 +117,6 @@ void mbedtls_ccm_free( mbedtls_ccm_context *ctx );
  * \param tag_len   The length of the tag to generate in Bytes:
  *                  4, 6, 8, 10, 12, 14 or 16.
  *
- * \note            The tag is written to a separate buffer. To concatenate
- *                  the \p tag with the \p output, as done in <em>RFC-3610:
- *                  Counter with CBC-MAC (CCM)</em>, use
- *                  \p tag = \p output + \p length, and make sure that the
- *                  output buffer is at least \p length + \p tag_len wide.
- *
  * \return          \c 0 on success.
```

Comment:
Some function documentations such as this one only document the return value `0`, while others such as the one for `mbedtls_ccm_setkey` explicitly mention the case of failure with a line like `A cipher-specific error code on failure.` This should be uniformized across the entire documentation.

Reply:
Seconded. In fact this function can return either a cipher-specific error code or `MBEDTLS_ERR_CCM_BAD_INPUT` or `MBEDTLS_ERR_CCM_HW_ACCEL_FAILED`. I think detailing the error conditions would be out of scope of this pull request, but it would be best to change the descriptions to “a CCM or cipher-specific error code” or words to this effect.

Reply:
I agree, and have already raised this issue of inconsistency with Hanno and Janos. I think generalizing as Gilles suhhested is the wisest course.  However, "a CCM or cipher-specific" - isn't CCM a type of cipher? I think it is covered by "cipher specific", no?

Reply:
CCM is a _mode_, not a _cipher_. The word _cipher_ can mean either the “core” algorithm such as DES, AES, Camellia, etc. that can only operate on fixed-size data, or the combination of a core algorithm and a _mode_ which specifies how to apply the core algorithm to arbitrary data size. (Even technical jargon is subject to metonymy.) In this context, “cipher-specific error code” refers to error codes from the core algorithm: `MBEDTLS_ERR_AES_XXX`, `MBEDTLS_ERR_CAMELLIA_XXX`, etc.

The CCM functions call the AES/Camellia/… functions under the hood, so they all potentially propagate AES/Camellia/… errors. “Potentially” because in some cases, the default software implementation doesn't cause an error, but an alternative implementation might.

Reply:
accepted and modified all to match